sqlglot/optimizer/qualify_columns.py

```diff
@@ -450,10 +450,16 @@ def _add_replace_columns(
         replace_columns[id(table)] = columns
 
 
-def _qualify_outputs(scope: Scope) -> None:
+def qualify_outputs(scope_or_expression: Scope | exp.Expression) -> None:
     """Ensure all output columns are aliased"""
-    new_selections = []
+    if isinstance(scope_or_expression, exp.Expression):
+        scope = build_scope(scope_or_expression)
+        if not isinstance(scope, Scope):
+            return
+    else:
+        scope = scope_or_expression
```

Comment:
Figured I'd make this accessible as a public helper of the `qualify_columns` module because it might be helpful elsewhere as well.